homeassistant/components/lifx/coordinator.py

```diff
@@ -169,32 +169,42 @@ def async_get_entity_id(self, platform: Platform, key: str) -> str | None:
     async def _async_update_data(self) -> None:
         """Fetch all device data from the api."""
         async with self.lock:
-            if self.device.host_firmware_version is None:
-                self.device.get_hostfirmware()
-            if self.device.product is None:
-                self.device.get_version()
-            if self.device.group is None:
-                self.device.get_group()
-
-            response = await async_execute_lifx(self.device.get_color)
-
-            if self.device.product is None:
-                raise UpdateFailed(
-                    f"Failed to fetch get version from device: {self.device.ip_addr}"
-                )
+            try:
+                await self._async_update_data_locked()
+            except asyncio.TimeoutError:
+                self.connection.async_stop()
+                self.connection.async_setup()
+            await self._async_update_data_locked()
```

Comment:
This would run regardless of whether the update succeeded or not, wouldn't it? Which would double the amount of requests to each device.

Reply:
Not intentional.  Indent fail